lib/maths/unittest/CTreeShapFeatureImportanceTest.cc

```diff
@@ -545,4 +559,38 @@ BOOST_FIXTURE_TEST_CASE(testSingleTreeShapRandomDataFrame, SFixtureSingleTreeRan
     });
 }
 
+BOOST_FIXTURE_TEST_CASE(testThreadedRandomTreeShap, SFixtureRandomTrees) {
+
+    // Test that threaded and non-threaded implementations agree.
+
+    core::startDefaultAsyncExecutor();
+
+    s_Frame->readRows(1, [&](TRowItr beginRows, TRowItr endRows) {
+        TSizeVec expectedIndices;
+        TStrVec expectedNames;
+        TVectorVec expectedShap;
+        for (auto row = beginRows; row != endRows; ++row) {
+            s_ForestFeatureImportance->shap(*row, [&](const TSizeVec& indices,
+                                                      const TStrVec& names,
+                                                      const TVectorVec& shap) {
+                expectedIndices = indices;
+                expectedNames = names;
+                expectedShap = shap;
+            });
+            s_ThreadedForestFeatureImportance->shap(*row, [&](const TSizeVec& indices,
+                                                              const TStrVec& names,
+                                                              const TVectorVec& shap) {
+                BOOST_REQUIRE_EQUAL(core::CContainerPrinter::print(expectedIndices),
+                                    core::CContainerPrinter::print(indices));
+                BOOST_REQUIRE_EQUAL(core::CContainerPrinter::print(expectedNames),
+                                    core::CContainerPrinter::print(names));
+                BOOST_REQUIRE_EQUAL(core::CContainerPrinter::print(expectedShap),
```

Comment:
Why is it necessary to compare the string representatives instead of the numerical values?

Reply:
It was just for code brevity, i.e. it allows me to compare all values (to reasonable accuracy) in one line.